tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -545,139 +545,105 @@ class TestLinearOperator(unittest.TestCase):
+        # defining the user-defined classes
+        class BaseMatlike(sp.linalg.LinearOperator):
 
-            # defining the sub-class
-            class HasAdjoint(BaseMatlike):
-                args = ()
+            def __init__(self):
+                self.dtype = A.dtype
+                self.shape = A.shape
 
-                def _adjoint(self):
-                    shape = self.shape[1], self.shape[0]
-                    matvec = partial(rmv, dtype=self.dtype)
-                    rmatvec = partial(mv, dtype=self.dtype)
-                    return sp.linalg.LinearOperator(matvec=matvec,
-                                                    rmatvec=rmatvec,
-                                                    dtype=self.dtype,
-                                                    shape=shape)
-            return (HasAdjoint(dtype), original)
+            def _adjoint(self):
+                shape = self.shape[1], self.shape[0]
+                return sp.linalg.LinearOperator(
+                    matvec=rmv, rmatvec=mv, dtype=self.dtype, shape=shape)
 
-        # defining the sub-classes
-        class HasRmatvec(BaseMatlike):
-            args = ()
+        class HasMatvec(BaseMatlike):
 
-            def _rmatvec(self, x):
-                return rmv(x, self.dtype)
+            def _matvec(self, x):
+                return mv(x)
 
-        class HasRmatmat(HasRmatvec):
+        class HasMatmat(BaseMatlike):
 
             def _matmat(self, x):
-                return original.dot(x)
-
-            def _rmatmat(self, x):
-                return original.T.conj().dot(x)
-
-        if(self.inner_modification == 'adjoint-matrix-vector'):
-            return (HasRmatvec(dtype), original)
-        if(self.inner_modification == 'adjoint-matrix-matrix'):
-            return (HasRmatmat(dtype), original)
-
-    def _outer_cases(self, xp, sp):
-        if(self.dtype != xp.complex_):
-            original = xp.array([[1., 2., 3.], [4., 5., 6.]])
-        else:
-            original = xp.array([[1, 2j, 3j], [4j, 5j, 6]])
-        if(self.outer_modification == 'normal'):
-            M, A = self._inner_cases(original, xp, sp, self.dtype)
-            return (sp.linalg.aslinearoperator(M), A)
-        if(self.outer_modification == 'transpose'):
-            M, A = self._inner_cases(original.T, xp, sp, self.dtype)
-            return (sp.linalg.aslinearoperator(M).T, A.T)
-        if(self.outer_modification == 'hermitian'):
-            M, A = self._inner_cases(original.T, xp, sp, self.dtype)
-            return (sp.linalg.aslinearoperator(M).H, A.T.conj())
-
-    @testing.numpy_cupy_array_equal(sp_name='sp')
+                return mv(x)
+
+        if self.inner_modification == 'normal':
+            return sp.linalg.aslinearoperator(A)
+        if self.inner_modification == 'sparse':
+            # TODO(asi1024): Fix to return contiguous matrix.
+            return sp.linalg.aslinearoperator(sp.csr_matrix(A))
```

Comment:
Added `contiguous_check=False` option, because somehow cupyx.scipy does not return C-contiguous matrix whereas scipy does.